sync/statesync/code_queue.go

```diff
@@ -61,19 +59,17 @@ func WithCapacity(n int) CodeQueueOption {
 // The `quit` channel, if non-nil, MUST eventually be closed to avoid leaking a
 // goroutine.
 func NewCodeQueue(db ethdb.Database, quit <-chan struct{}, opts ...CodeQueueOption) (*CodeQueue, error) {
-	// Apply defaults then options.
-	o := queueConfig{
+	// Create with defaults, then apply options.
+	q := &CodeQueue{
+		db:       db,
+		quit:     quit,
 		capacity: defaultQueueCapacity,
 	}
-	options.ApplyTo(&o, opts...)
+	options.ApplyTo(q, opts...)
 
-	ch := make(chan common.Hash, o.capacity)
-	q := &CodeQueue{
-		db:   db,
-		in:   ch,
-		out:  ch,
-		quit: quit,
-	}
+	ch := make(chan common.Hash, q.capacity)
+	q.in = ch
+	q.out = ch
```

Comment:
(nit) Declare `ch` before `q` and then just set the fields in the literal, like it was before.